ServerlessLibraryUI/src/components/AppTopbar.vue

```diff
@@ -79,13 +81,24 @@ export default {
 
     getDataToLog() {
       return this.activeFilters;
+    },
+
+    setFilter()
+    {
+      this.searchtext= this.filters.filtertext;
+      this.language = this.filters.language;
+      this.type = this.filters.type;
     }
   },
 
   watch: {
     activeFilters() {
       this.$emit('updateFilters', this.activeFilters)
     },
+    filters()
```

Comment:
why is a watch required for 'filters'? how does it get triggered when any of the search criteria is changed?